src/OSPSuite.Core/Domain/ParameterIdentifications/WeightedObservedData.cs

```diff
@@ -24,8 +24,17 @@ public WeightedObservedData(DataRepository observedData)
       public static implicit operator DataRepository(WeightedObservedData weightedObservedData) => weightedObservedData?.ObservedData;
 
       public string Name => ObservedData?.Name;
+
+      /// <summary>
+      ///    Identifier used to uniquely identify a weighted observed data. The same observed data could be used more than
+      ///    once in a mapping
+      /// </summary>
+      public int? Id { get; set; }
```

Comment:
That's my trick for display name unicity. An Id that is set everytime multiple instances of the same observed data is used
Default is null so that we are compatible with previous PR
if we rename the obs data, we do not need to do anythig